staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,297 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// These are valid flow-dingtinguisher methods.
+const (
+	// FlowDistinguisherMethodByUser specifies that the flow distinguisher is the username works in the request context
+	// so that the requests from the same user will enqueued into the same set of queues for processing.
+	FlowDistinguisherMethodByUserType FlowDistinguisherMethodType = "ByUser"
+
+	// FlowDistinguisherMethodByNamespace works by computing flow distinguisher with the user requested namespace. Any
+	// user requesting the namespace will be put into the same queue for processing.
+	// FlowDistinguisherMethodByNamespaceType specifies that the flow distinguisher is the namespace of the object
+	// that the request accesses.  If the object is not namespaced or the request is a non-resource request then
+	// the flow distinguisher is the empty string.
+	FlowDistinguisherMethodByNamespaceType FlowDistinguisherMethodType = "ByNamespace"
+
+	GroupKind          = "Group"
+	ServiceAccountKind = "ServiceAccount"
+	UserKind           = "User"
+
+	APIGroupAll    = "*"
+	ResourceAll    = "*"
+	VerbAll        = "*"
+	NonResourceAll = "*"
+	NameAll        = "*"
+)
+
+// System preset priority level names
+const (
+	PriorityLevelConfigurationNameExempt = "exempt"
+)
+
+// Conditions
+const (
+	FlowSchemaConditionDangling = "Dangling"
+
+	PriorityLevelConfigurationConditionConcurrencyShared = "ConcurrencyShared"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// FlowSchema defines the schema of a group of flows. Note that a flow makes up of a set of inbound API requests with
```

Comment:
This comment won't help folks that don't already know what is meant by flow and schema.

s/flow makes up/flow is made up/

Reply:
You're right, the first sentence is just a tautology.  OTOH, the second sentence attempts to define "flow".  And yes, your replacement looks like an improvement to me.

Reply:
fixed in #81788

Reply:
We should still replace the tautology.  Perhaps something like "A flow schema defines a set of flows and assigns them to a priority level".